rspace/src/main/scala/coop/rchain/rspace/channelStore/instances/ChannelStoreImpl.scala

```diff
@@ -53,6 +53,39 @@ object ChannelStoreImpl {
       } yield ()
 
     override def getChannelHash(hash: Blake2b256Hash): F[Option[ChannelHash]] = store.get(hash)
+
+    override def putChannelHashes(channels: Seq[C]): F[Unit] = {
+      def convert(channel: C): F[(Blake2b256Hash, DataJoinHash)] =
+        for {
+          // C => hash(C)
+          eventKey <- Sync[F].delay(StableHashProvider.hash(channel)(sc))
+
+          // C => hashPrefix(C)
+          dataHash = hashDataChannel(channel, sc)
+          joinHash = hashJoinsChannel(channel, sc)
+        } yield (eventKey, DataJoinHash(dataHash, joinHash))
+
+      channels.toList.traverse(convert).flatMap(store.put)
+    }
+
+    override def putContinuationHashes(conts: Seq[Seq[C]]): F[Unit] = {
+      def convert(channels: Seq[C]): F[(Blake2b256Hash, ContinuationHash)] =
+        for {
+          // Hash each channel in a list
+          channelsHashes <- Sync[F].delay(
+                             toOrderedByteVectors(channels)(sc).map(Blake2b256Hash.create)
+                           )
+          // Concatenate channel hashes and hash result
+          // Seq[C] => Seq[hash(C)] => Seq[bytes(C)] => bytes(C) => hash(C)
+          eventKey = continuationKey(channelsHashes)
+
+          // Concatenate channels and hash result
+          // Seq[C] => Seq[bytes(C)] => bytes(C) => hashPrefix(C)
+          continuationHash = hashContinuationsChannels(channels, sc)
+        } yield (eventKey, ContinuationHash(continuationHash))
+
+      conts.toList.traverse(convert).flatMap(store.put)
```

Comment:
Sorted keys can also speed up insertion.
https://stackoverflow.com/a/62980373

In LMDB by default the keys are compared lexically. ByteString has also lexical ordering defined. Maybe it's worth to test.
https://stackoverflow.com/questions/39387702/sorting-an-lmdb-file-for-sequential-access-according-to-key-order
http://104.237.133.194/doc/group__mdb.html#ga68e47ffcf72eceec553c72b1784ee0fe